### PR TITLE
Add decodedAsBase64 to AbstractStringAssert

### LINKED ISSUE #1805: Base64 support for AbstractStringAssert
#### Summary
Java 8 has built-in support for Base64 with `java.util.Base64`.

What about adding decoding assertions to `AbstractStringAssert`?

#### Example

```java
assertThat("THVrZSBTa3l3YWxrZXI=")
  .isValidBase64()  // relies on java.util.Base64.getDecoder().decode(String) which throws IllegalArgumentException in case of wrong input
  .decodedAsBase64() // Changes the object under assertion, returning a new ByteArrayAssert
  ... // additional byte array assertions
```

### PR DESCRIPTION
**Pull request by scordio**
#### Check List:
* Fixes #1805
* Unit tests : YES
* Javadoc with a code example (on API only) : YES

### CODE REVIEW

File: src/main/java/org/assertj/core/api/AbstractStringAssert.java [hunk 1]
```diff
@@ -242,6 +243,31 @@ public SELF isValidBase64() {
     return myself;
   }
 
+  /**
+   * Decodes the actual value as a Base64 encoded string, the decoded value becoming the new object under test.
+   * <p>
+   * Examples:
+   * <pre><code class='java'> // assertions succeeds
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/AbstractStringAssert.java:250]: assertions -> assertion

File: src/main/java/org/assertj/core/api/AbstractStringAssert.java [hunk 2]
```diff
@@ -242,6 +243,31 @@ public SELF isValidBase64() {
     return myself;
   }
 
+  /**
+   * Decodes the actual value as a Base64 encoded string, the decoded value becoming the new object under test.
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/AbstractStringAssert.java:247]: `the decoded value becoming ` -> `the decoded value (a byte[]) becoming `

Comment (scordio) [reply src/main/java/org/assertj/core/api/AbstractStringAssert.java:247]: What about `the decoded bytes becoming the new array under test`?

**Comment by joel-costigliola** [comment]
Integrated thanks @scordio!

**Pull request merged**